tests/unit/lib/FluxibleContext.js

```diff
@@ -441,6 +513,81 @@ describe('FluxibleContext', function () {
+                            done(err);
+                        }
+
+                        expect(actionCalls.length).to.equal(4);
+                        expect(actionCalls[0].context).to.contain.keys(Object.keys(actionContext));
+                        expect(actionCalls[0].context).to.contain.keys(['id','stack']);
+                        expect(actionCalls[0].context.id).to.equal('THIS_IS_UNIQUE');
+                        expect(actionCalls[0].context.stack.join('.')).to.equal('Action.One');
+                        expect(actionCalls[0].payload).to.equal(payload);
+                        expect(actionCalls[1].context).to.contain.keys(Object.keys(actionContext));
+                        expect(actionCalls[1].context).to.contain.keys(['id','stack']);
+                        expect(actionCalls[1].context.id).to.equal('THIS_IS_UNIQUE');
+                        expect(actionCalls[1].context.stack.join('.')).to.equal('Action.One.Two');
+                        expect(actionCalls[1].payload).to.equal(payload);
+                        expect(actionCalls[2].context).to.contain.keys(Object.keys(actionContext));
+                        expect(actionCalls[2].context).to.contain.keys(['id','stack']);
+                        expect(actionCalls[2].context.id).to.equal('THIS_IS_UNIQUE');
+                        expect(actionCalls[2].context.stack.join('.')).to.equal('Action.One.Two');
+                        expect(actionCalls[2].payload).to.equal(payload);
+                        expect(actionCalls[3].context).to.contain.keys(Object.keys(actionContext));
+                        expect(actionCalls[3].context).to.contain.keys(['id','stack']);
+                        expect(actionCalls[3].context.id).to.equal('THIS_IS_UNIQUE');
+                        expect(actionCalls[3].context.stack.join('.')).to.equal('Action.One.Three');
+                        expect(actionCalls[3].payload).to.equal(payload);
+                        done();
+                    });
+                };
+                actionOne.displayName = 'One';
+                var actionTwo = function (context, payload, callback) {
+                    actionCalls.push({
+                        context: context,
+                        payload: payload
+                    });
+                    callback();
+                };
+                actionTwo.displayName = 'Two';
+                var actionThree = function (context, payload, callback) {
+                    actionCalls.push({
+                        context: context,
+                        payload: payload
+                    });
+                    callback();
+                };
+                actionThree.displayName = 'Three';
+                var payload = {};
+                // delete componentContext cache to regenerate it with id this time
+                delete context._componentContext
+                componentContext = context.getComponentContext({
+                    id: 'THIS_IS_UNIQUE'
```

Comment:
`getComponentContext` is really only called once on the client so maybe it makes sense to pass in a unique id when calling `executeAction` instead.